### ISSUE TITLE
Sudden drop in accuracy

### ISSUE THREAD

**Issue by wing212** [voldemortX/DST-CBC#12]
Hello, I want to ask why the accuracy has suddenly dropped, and the accuracy of my reproduced article is much lower than that of the original text. I use a single 3090ti graphics card for training.
![image](https://user-images.githubusercontent.com/68488554/144184851-c5f73f2a-897f-4f57-a1e2-2b4bfda397fc.png)

**Comment by voldemortX** [comment]
@wing212 Hi! Did you run code from the latest master branch? And could I ask what is your exact setup? e.g. PyTorch & TorchVision version, CUDA version, etc.

Btw, I don't think there is 3090ti in the market yet, did you mean 3090?

**Comment by wing212** [comment]
Yes, I made a mistake。it's 309
![RIXX7V9B3)0M 5D0EXAE_RB](https://user-images.githubusercontent.com/68488554/144186380-bbddf473-39b0-4c0e-97f7-e8e2be2149d9.png)
![E$ }_00Q(2_K~4 E@FX8Y7F](https://user-images.githubusercontent.com/68488554/144186383-67e6f3c3-25c3-4adc-860e-3f47e1d5337e.png)
0。

> @wing212你好！您是否从最新的主分支运行代码？我能问一下你的确切设置是什么吗？例如 PyTorch & TorchVision 版本、CUDA 版本、是否使用混合精度等。
> 
> 顺便说一句，我认为市场上还没有3090ti，你的意思是3090吗？

**Comment by wing212** [comment]
Yes, I made a mistake。it's 309
![RIXX7V9B3)0M 5D0EXAE_RB](https://user-images.githubusercontent.com/68488554/144186380-bbddf473-39b0-4c0e-97f7-e8e2be2149d9.png)
![E$ }_00Q(2_K~4 E@FX8Y7F](https://user-images.githubusercontent.com/68488554/144186383-67e6f3c3-25c3-4adc-860e-3f47e1d5337e.png)
0。

> @wing212你好！您是否从最新的主分支运行代码？我能问一下你的确切设置是什么吗？例如 PyTorch & TorchVision 版本、CUDA 版本、是否使用混合精度等。
> 
> 顺便说一句，我认为市场上还没有3090ti，你的意思是3090吗？

**Comment by voldemortX** [comment]
Did you accidentally closed the issue?

**Comment by wing212** [comment]
> @wing212 Hi! Did you run code from the latest master branch? And could I ask what is your exact setup? e.g. PyTorch & TorchVision version, CUDA version, whether mixed precision is used, etc.
> 
> Btw, I don't think there is 3090ti in the market yet, did you mean 3090?
It is 3090. I have used ape
![RIXX7V9B3)0M 5D0EXAE_RB](https://user-images.githubusercontent.com/68488554/144186652-46b471ff-ff90-44ab-8683-2725af778870.png)
![E$ }_00Q(2_K~4 E@FX8Y7F](https://user-images.githubusercontent.com/68488554/144186656-f8c62bbc-8d9a-44a6-85bd-8196d815c740.png)
x.

**Comment by wing212** [comment]
> @wing212 Hi! Did you run code from the latest master branch? And could I ask what is your exact setup? e.g. PyTorch & TorchVision version, CUDA version, whether mixed precision is used, etc.
> 
> Btw, I don't think there is 3090ti in the market yet, did you mean 3090?

should I  adjust my learning rate?

**Comment by voldemortX** [comment]
The performance should be reproduceable from my provided shells. Although if you use different PyTorch versions and NVIDIA's apex, it could lead to problems like gradient explosions. Could you give me more details about your shell commands and code version? Instead of using a different learning rate, I recommend full precision training first to isolate the problem.

**Comment by wing212** [comment]
Yes, there are problems with the baseline, I don't know why. I am using the code of the single card training version
![}52OQL(QD~9II{MAX~W(2AU](https://user-images.githubusercontent.com/68488554/144188670-9e87d5aa-b6af-45e4-a6bd-8d1f5963fc9d.png)
![`X5NW_)O~S1QR1JK `_{RRE](https://user-images.githubusercontent.com/68488554/144188719-3d4539a7-daa3-4eb8-95bd-96ed6ead5efa.png)
.

**Comment by voldemortX** [comment]
@wing212 Firstly, if you are using PyTorch >= 1.6, You best use the master branch. Also a clear environment without the standalone apex package. I suspect the problem is from there. Maybe refer to #2 and check your loss in tensorboard.

Secondly, for 1/30 cityscapes baselines, the number of epochs should be longer. See Appendix B in the paper. Although yours seems good enough there, we still should align the settings.

**Comment by wing212** [comment]
> @wing212首先，如果您使用的是 PyTorch >= 1.6，则最好使用主分支。也是一个没有独立顶点包的清晰环境。我怀疑问题出在哪里。也许参考#2并检查你在张量板中的损失。
> 
> 其次，对于1/30的城市景观基线，纪元的数量应该更长。见文件附录B。虽然你的看起来足够好，但我们仍然应该对齐设置。
> 
> [重要]你对城市景观的了解似乎不对吗？它应为表 1 中的 0.004。
I originally wanted to try training with other learning rates to see what the results are. I forgot to change the learning rate to 0.001. I will change the epoch to 329 without using apex.

**Comment by voldemortX** [comment]
@wing212 Great! Let me know if you still can't reproduce results. FYI, the std for Cityscapes 1/30 experiments is 2.56 in my records.

**Comment by wing212** [comment]
> @wing212伟大！如果您仍然无法重现结果，请告诉我。仅供参考，在我的记录中，Cityscapes 1/30实验的标准是2.56。

Thank you, I will keep trying. I would like to ask what is 2.56?

**Comment by voldemortX** [comment]
> > @wing212伟大！如果您仍然无法重现结果，请告诉我。仅供参考，在我的记录中，Cityscapes 1/30实验的标准是2.56。
> 
> Thank you, I will keep trying. I would like to ask what is 2.56?

STD (标准差)

**Comment by wing212** [comment]
> > > @wing212 great! If you still can't reproduce the results, please let me know. For informational purposes only, the standard for the Cityscapes 1/30 experiment is 2.56 in my records.
> > 
> > 
> > Thank you, I will keep trying. I would like to ask what is 2.56?
> 
> STD (STANDARD DIFFERENCE)

ok

**Comment by wing212** [comment]
Sorry to interrupt you again, I did not use mixed-precision this time, but the result is still wrong. The  1/50 challenge of VOC data set is very low accuracy, and the 1/30 challenge  of cityscapes data set is abnormally high
![S%GABQ4600I8 5J _K5E8)A](https://user-images.githubusercontent.com/68488554/144337958-31291570-c00f-4478-a6c3-aa73c38ea14d.png)
![U@{5K0D$THLP XXR`M92)}1](https://user-images.githubusercontent.com/68488554/144337970-8aa51984-42b4-46c3-b303-6dc18baa015e.png)
![S`3HOL9L{36K1TDJ1R15CZ2](https://user-images.githubusercontent.com/68488554/144337985-c821c44b-eb10-40b0-9019-3abe92d50fd5.png)

![6@AW`}AP5JHR 42`H8YP@VX](https://user-images.githubusercontent.com/68488554/144337998-0f73affb-02a7-4daa-b93e-5ed957e98d27.png)

.

**Comment by wing212** [comment]
![U@{5K0D$THLP XXR`M92)}1](https://user-images.githubusercontent.com/68488554/144338162-507defcf-1883-4c73-9c02-4e56a745ef02.png)

**Comment by voldemortX** [comment]
@wing212 Actually, your results look fine. I did get a 57.24 for city-30-1, that is why we need to take averages in these experiments (the STD is quite large). My voc-50-1 was 63.50 (it was weirdly low but I kept it to be fair), try 50-0 or 50-2 the results should be higher and give you an average around 67.

**Comment by wing212** [comment]
> @wing212 Actually, your results look fine. I did get a 57.24 for city-30-1, that is why we need to take averages in these experiments (the STD is quite large). My voc-50-1 was 63.50 (it was weirdly low but I kept it to be fair), try 50-0 or 50-2 the results should be higher and give you an average around 67.

I will continue， thank you.

**Comment by voldemortX** [comment]
But it looks like you have sid=2 for the voc experiment, do check the dataset and data splits if you can't get similar avg from 3 runs.

**Comment by wing212** [comment]
> 但看起来 voc 实验的 sid=2，如果无法从 3 次运行中获得类似的平均值，请检查数据集和数据拆分。

Ok， I will do it.
